homeassistant/components/zwave_js/__init__.py

```diff
@@ -624,6 +629,29 @@ def register_node_in_dev_reg(self, node: ZwaveNode) -> dr.DeviceEntry:
         else:
             ids = {device_id}
 
+        # Check if this node was previously provisioned
+        provisioning_entry = await controller.async_get_provisioning_entry(node.node_id)
+
+        if (
+            provisioning_entry
+            and hasattr(provisioning_entry, "device_id")
+            and provisioning_entry.device_id
+        ):
+            preprovisioned_device = self.dev_reg.async_get(provisioning_entry.device_id)
+
+            if preprovisioned_device:
+                dsk = provisioning_entry.dsk
+                dsk_identifier = (DOMAIN, f"provision_{dsk}")
+
+                # If the pre-provisioned device has the DSK identifier, remove it
+                if dsk_identifier in preprovisioned_device.identifiers:
+                    new_identifiers = preprovisioned_device.identifiers.copy()
+                    new_identifiers.remove(dsk_identifier)
+                    new_identifiers.update(ids)
+                    self.dev_reg.async_update_device(
+                        preprovisioned_device.id, new_identifiers=new_identifiers
+                    )
+
         device = self.dev_reg.async_get_or_create(
```

Comment:
Should we still try to create a new device if we've updated the existing device with the node id above?

Reply:
You're right. I'll just add the other stuff to the device update and return early